components/Pages/Dashboard/hooks/useCurrentEpoch.ts

```diff
@@ -0,0 +1,69 @@
+import {Wallet} from "../../../../util/wallet-adapters";
+import {
+  FEE_DISTRIBUTOR_CONTRACT_ADDRESS
+} from "../../../../constants/bonding_contract";
+import {JsonObject} from "@cosmjs/cosmwasm-stargate";
+import {useQuery} from "react-query";
+
+interface EpochData {
+  id: string;
+  start_time: string;
+  total: {
+    amount: string;
+    info: {
+      native_token: {
+        denom: string;
+      };
+    };
+  }[];
+  available: {
+    amount: string;
+    info: {
+      native_token: {
+        denom: string;
+      };
+    };
+  }[];
+  claimed: {
+    amount: string;
+    info: {
+      native_token: {
+        denom: string;
+      };
+    };
+  }[];
+}
+interface Epoch {
+  epoch: EpochData;
+}
+export const useCurrentEpoch = (client: Wallet | null) => {
```

Comment:
I like how you be splitting the queries into different hooks. Maybe if we get loads of different types of hooks we could segregate it even more. But this looks great